net/src/p2p_config.rs

```diff
@@ -0,0 +1,147 @@
+use holochain_core_types::{error::HolochainError, json::JsonString};
+use std::{fs::File, str::FromStr};
+
+//--------------------------------------------------------------------------------------------------
+// P2pBackendKind
+//--------------------------------------------------------------------------------------------------
+
+#[derive(Deserialize, Serialize, Clone, Debug, DefaultJson, PartialEq, Eq)]
+pub enum P2pBackendKind {
+    MOCK,
+    IPC,
+}
+
+impl FromStr for P2pBackendKind {
+    type Err = ();
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "MOCK" => Ok(P2pBackendKind::MOCK),
+            "IPC" => Ok(P2pBackendKind::IPC),
+            _ => Err(()),
+        }
+    }
+}
+
+impl From<P2pBackendKind> for String {
+    fn from(kind: P2pBackendKind) -> String {
+        String::from(match kind {
+            P2pBackendKind::MOCK => "MOCK",
+            P2pBackendKind::IPC => "IPC",
+        })
+    }
+}
+
+impl From<String> for P2pBackendKind {
+    fn from(s: String) -> P2pBackendKind {
+        P2pBackendKind::from_str(&s).expect("could not convert String to P2pBackendKind")
+    }
+}
+
+impl From<&'static str> for P2pBackendKind {
+    fn from(s: &str) -> P2pBackendKind {
+        P2pBackendKind::from(String::from(s))
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+// P2pConfig
+//--------------------------------------------------------------------------------------------------
+
+#[derive(Deserialize, Serialize, Clone, Debug, DefaultJson, PartialEq)]
+pub struct P2pConfig {
+    pub backend_kind: P2pBackendKind,
+    pub backend_config: serde_json::Value,
+}
+
+// Conversions
+impl FromStr for P2pConfig {
+    type Err = String;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        serde_json::from_str(s).map_err(|err| err.to_string())
+    }
+}
+impl P2pConfig {
+    pub fn as_str(&self) -> String {
+        serde_json::to_string(self).unwrap()
```

Comment:
`expect(...)` here instead of `unwrap()`? Or better yet, use TryFrom instead, so it can return a result?

Reply:
unwrap() is fine since there is no way this can fail since this struct derives from Serialize.
Would you want me to add that as a comment?

Reply:
fixed

Reply:
oh, cool :+1: - yeah comment would be good, my brain is trained to think "unwrap bad" outside of tests : )

Reply:
Only time unwrap is fine is when code logic makes it impossible to fail, like here :)